src/androidTest/java/com/couchbase/lite/replicator/ReplicationTest.java

```diff
@@ -5121,4 +5121,118 @@ public boolean run() {
+            MockBulkDocs mockBulkDocs = new MockBulkDocs();
+            mockBulkDocs.setSticky(true);
+            dispatcher.enqueueResponse(MockHelper.PATH_REGEX_BULK_DOCS, mockBulkDocs);
+
+            // checkpoint PUT response (sticky)
+            MockCheckpointPut mockCheckpointPut = new MockCheckpointPut();
+            mockCheckpointPut.setSticky(true);
+            dispatcher.enqueueResponse(MockHelper.PATH_REGEX_CHECKPOINT, mockCheckpointPut);
+
+            // 2. Kick off continuous push replication
+            Replication replicator = database.createPushReplication(server.getUrl("/db"));
+            replicator.setContinuous(true);
+            CountDownLatch replicationIdleSignal = new CountDownLatch(1);
+            ReplicationIdleObserver replicationIdleObserver = new ReplicationIdleObserver(replicationIdleSignal);
+            replicator.addChangeListener(replicationIdleObserver);
+            replicator.start();
+
+            // 3. Wait for document to be pushed
+
+
+            // NOTE: (Not 100% reproducible) With CBL Java on Jenkins (Super slow environment),
+            //       Replicator becomes IDLE between batches for this case, after 100 push replicated.
+            // TODO: Need to investigate
+
+            // wait until replication goes idle
+            boolean successful = replicationIdleSignal.await(60, TimeUnit.SECONDS);
+            assertTrue(successful);
+
+            // wait until mock server gets the checkpoint PUT request
+            boolean foundCheckpointPut = false;
+            String expectedLastSequence = String.valueOf(NUM_DOCS);
+            while (!foundCheckpointPut) {
+                RecordedRequest request = dispatcher.takeRequestBlocking(MockHelper.PATH_REGEX_CHECKPOINT);
+                if (request.getMethod().equals("PUT")) {
+                    foundCheckpointPut = true;
+                    String body = request.getUtf8Body();
+                    Log.e("testExcessiveCheckpointingDuringPushReplication", "body => " + body);
+                    // TODO: this is not valid if device can not handle all replication data at once
+                    if (System.getProperty("java.vm.name").equalsIgnoreCase("Dalvik")) {
+                        assertTrue(body.indexOf(expectedLastSequence) != -1);
+                    }
+                    // wait until mock server responds to the checkpoint PUT request
+                    dispatcher.takeRecordedResponseBlocking(request);
+                }
+            }
+
+            // make some assertions about the outgoing _bulk_docs requests
+            RecordedRequest bulkDocsRequest1 = dispatcher.takeRequest(MockHelper.PATH_REGEX_BULK_DOCS);
+            assertNotNull(bulkDocsRequest1);
+
+            if (System.getProperty("java.vm.name").equalsIgnoreCase("Dalvik")) {
+
+                // TODO: Need to fix
+                RecordedRequest bulkDocsRequest2 = dispatcher.takeRequest(MockHelper.PATH_REGEX_BULK_DOCS);
```

Comment:
In stead of fixing them now, I created new ticket for this. https://github.com/couchbase/couchbase-lite-java-core/issues/871